compiler/rustc_hir_typeck/src/closure.rs

```diff
@@ -632,8 +633,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // function.
                 Some(hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Fn)) => {
                     debug!("closure is async fn body");
-                    self.deduce_future_output_from_obligations(expr_def_id, body.id().hir_id)
-                        .unwrap_or_else(|| {
+                    let def_id =
+                        self.tcx.hir().opt_local_def_id(body.id().hir_id).unwrap_or(CRATE_DEF_ID);
```

Comment:
```suggestion
                        self.tcx.hir().body_owner_def_id(body.id());
```
I think that value is actually the same as `expr_def_id` (which is always local btw).